tests/integration-tests/clusters_factory.py

```diff
@@ -187,25 +188,38 @@ def describe_cluster(self):
             logging.error("Failed when getting cluster status with error:\n%s\nand output:\n%s", e.stderr, e.stdout)
             raise
 
-    def instances(self, desired_instance_role=None):
+    def describe_compute_fleet(self):
+        """Run pcluster describe-compute-fleet and return the result."""
+        cmd_args = ["pcluster", "describe-compute-fleet", "--cluster-name", self.name]
+        try:
+            result = run_command(cmd_args, log_error=False)
+            response = json.loads(result.stdout)
+            logging.info("Describe cluster %s compute fleet successfully", self.name)
+            return response
+        except subprocess.CalledProcessError as e:
+            logging.error(
+                "Failed when getting cluster compute fleet with error:\n%s\nand output:\n%s", e.stderr, e.stdout
+            )
+            raise
+
+    def instances(self, node_type=None, queue_name=None):
```

Comment:
I think we might want to update this function name to more directly represent the underlying call that it makes to pcluster (e.g. "describe_cluster_instances")... though a bit verbose I think it's better to be consistent / explicit.

Reply:
Looking further down, I'm actually curious if it's better to have it just return the instances (e.g. `return response["instances"]`) or if it should be called `get_cluster_incance_ids` to reflect that it is just returning the ids?

Reply:
Done by renaming it to get_cluster_instance_ids.